services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyHttpHandlerHelper.java

```diff
@@ -46,53 +60,53 @@
                 } else {
-                    s_logger.error("decode token. host info is not found!");
+                    logMessageIfCannotFindClientParam(HOST);
                 }
                 if (param.getClientHostPort() != 0) {
                     s_logger.debug("decode token. port: " + param.getClientHostPort());
-                    map.put("port", String.valueOf(param.getClientHostPort()));
+                    map.put(PORT, String.valueOf(param.getClientHostPort()));
                 } else {
-                    s_logger.error("decode token. port info is not found!");
+                    logMessageIfCannotFindClientParam(PORT);
                 }
                 if (param.getClientTag() != null) {
                     s_logger.debug("decode token. tag: " + param.getClientTag());
-                    map.put("tag", param.getClientTag());
+                    map.put(TAG, param.getClientTag());
                 } else {
-                    s_logger.error("decode token. tag info is not found!");
+                    logMessageIfCannotFindClientParam(TAG);
                 }
                 if (param.getClientHostPassword() != null) {
-                    map.put("sid", param.getClientHostPassword());
+                    map.put(SID, param.getClientHostPassword());
                 } else {
-                    s_logger.error("decode token. sid info is not found!");
+                    logMessageIfCannotFindClientParam(SID);
                 }
                 if (param.getClientTunnelUrl() != null)
-                    map.put("consoleurl", param.getClientTunnelUrl());
+                    map.put(CONSOLE_URL, param.getClientTunnelUrl());
                 if (param.getClientTunnelSession() != null)
-                    map.put("sessionref", param.getClientTunnelSession());
+                    map.put(SESSION_REF, param.getClientTunnelSession());
                 if (param.getTicket() != null)
-                    map.put("ticket", param.getTicket());
+                    map.put(TICKET, param.getTicket());
                 if (param.getLocale() != null)
-                    map.put("locale", param.getLocale());
+                    map.put(LOCALE, param.getLocale());
                 if (param.getHypervHost() != null)
-                    map.put("hypervHost", param.getHypervHost());
+                    map.put(HYPERV_HOST, param.getHypervHost());
                 if (param.getUsername() != null)
-                    map.put("username", param.getUsername());
+                    map.put(USERNAME, param.getUsername());
                 if (param.getPassword() != null)
-                    map.put("password", param.getPassword());
+                    map.put(PASSWORD, param.getPassword());
             } else {
-                s_logger.error("Unable to decode token");
+                s_logger.error("Unable to decode token due to null console proxy client param");
             }
         } else {
```

Comment:
This else statement seems redundant and can go if the tokenMap only contains 1 TOKEN item